rkt/metadata_service.go

```diff
@@ -541,12 +539,19 @@ func runMetadataService(args []string) (exit int) {
 
 	log.Print("Metadata service running...")
 
-	if err = srv.Serve(l); err != nil {
-		log.Printf("Error serving HTTP: %v", err)
+	go func() {
+		if err = srv.Serve(unixl); err != nil {
```

Comment:
This whole piece needs better handling of exit path and needs to process SIGTERM/SIGINT well. But that'll be a separate PR.